### DISCUSSION TITLE
Python support preparation

### DISCUSSION THREAD

**Discussion by sonalgoyal** [zinggAI/zingg#289] [Ideas]
As we are building some functionality in python, need to see what all support we need.

a. zingg.sh update
b. docker update?
c. python dependencies and thier installation and distribution through --py-files?
d. installation documentation update

**Comment by sonalgoyal** [comment]
Also need to see what all we can suppoort in python. first off, we should expose the model learning metrics as a new phase and use sklearn etc to demonstrate the learning so that users know how to stop. See #254

**Comment by sonalgoyal** [comment]
We should also support defining the data, output, field definitions and other config params so that people do not have to manually build the config.json.

**Comment by navinrathore** [comment]
Following are the next set of items planned:
- rename 'client' to 'zingg'
- All the methods of client (Python) should be defined in Java.
-- BuildAndSetArgument() function should be added
- similar to setData(), new function setOutput() shall be written
- FieldDefinition should take simple quotes as inputs. e.g. "string" rather than "\"string\"". Internally, needful shall be done
- New MatchType class in python.
- MatchType should be able to passed in FieldDefinition as VarArgs.
- ClientOptions - new class in Python
- new functions in zingg.py - getDFFromDS(), getPandasDFFromDS()
- Arguments python class should have new function writeArgumentsToJson() that will dump Argument object into a json file
- New functions of client.java (*Stat() and getMarkedRecords()) should be added into Python counterpart
- New getUnmarkedRecords()like that of marked Records
- Issue #296 AssessModel phase should be updated accordingly. All references of '_jvm' should be removed.
-  A new phase PeekModel should be introduced. this could be executed to get necessary basic data from Zingg
-- ZFactory shall have necessary change to return default phase 'PeekModel'
